src/types/utilities.d.ts

```diff
@@ -0,0 +1,5 @@
+type UnionKeys<T> = T extends T ? keyof T : never
+type StrictUnionHelper<T, TAll> = T extends unknown
+  ? T & Partial<Record<Exclude<UnionKeys<TAll>, keyof T>, never>>
+  : never
```

Comment:
🤯 I have to say, i didn't get how this work. 
I see it does though..

Reply:
Partial and Exclude AND UnionKeys 😂 yeah i have trouble reading this but if it works...? 🤷

Reply:
yeah... it would be great if TS already implemented this as a utility type